Tests/PointFreeTests/DiscountsTests.swift

```diff
@@ -29,15 +30,145 @@ class DiscountsTests: TestCase {
     )
   }
 
-  func testDiscounts_LoggedIn() {
+  func testDiscounts_LoggedIn_PercentOff_Forever() {
+    let fiftyPercentOffForever = Coupon(
+      duration: .forever,
+      id: "deadbeef",
+      name: "50% off forever",
+      rate: .percentOff(50),
+      valid: true
+    )
+    update(
+      &Current,
+      \.database.fetchSubscriptionById .~ const(pure(nil)),
+      \.database.fetchSubscriptionByOwnerId .~ const(pure(nil)),
+      \.stripe.fetchCoupon .~ const(pure(fiftyPercentOffForever))
+    )
+
+    assertSnapshot(
+      matching: connection(
+        from: request(with: secureRequest("http://localhost:8080/discounts/blobfest"), session: .loggedIn)
+        )
+        |> siteMiddleware,
+      as: .ioConn
```

Comment:
Worth adding image snaps as well? Or are you worried about repo size?

Reply:
it did seem like overkill, and we do have some coupon coverage in `SubscriptionConfirmationTests.swift`, just not on every permutation of rate and duration.